ui/apps/platform/src/hooks/useURLSort.ts

```diff
@@ -45,65 +34,161 @@ function tableSortOption(
+ *     - This signature is used when the column has multiple sort fields. In this form, the fields sent to
+ *     the API will be the `field` values in the `fieldOptions` array, each of which can optionally
+ *     include an aggregate. The correct column index in the PatternFly table is found by finding any array
+ *     provided in the `sortFields` option that contains one of the `field` values in the `fieldOptions` array.
+ *
+ * @example
+ * // A use case where a table only has single sort columns
+ * const { sortOption, setSortOption, getSortParams } = useURLSort({
+ *    sortFields: ['name', 'age', 'date'],
+ *    defaultSortOption: { field: 'name', direction: 'asc' },
+ * });
+ *
+ * return (
+ *   <Thead>
+ *     <Th sort={getSortParams('name')}>Name</Th>
+ *     <Th sort={getSortParams('age')}>Age</Th>
+ *     <Th sort={getSortParams('date')}>Date</Th>
+ *   </Thead>
+ * );
+ *
+ * @example
+ * // A use case where a table has both single and multi sort columns
+ * const { sortOption, setSortOption, getSortParams } = useURLSort({
+ *   sortFields: ['name', 'age', 'date', ['critical count', 'important count']],
+ *  defaultSortOption: { field: 'name', direction: 'asc' },
+ * });
+ *
+ * return (
+ *   <Thead>
+ *     <Th sort={getSortParams('name')}>Name</Th>
+ *     <Th sort={getSortParams('age')}>Age</Th>
+ *     <Th sort={getSortParams('date')}>Date</Th>
+ *    <Th sort={getSortParams('critical count', [
+ *       { field: 'critical count' },
+ *       { field: 'important count' },
+ *    ])}>Critical Count</Th>
+ *   </Thead>
+ * );
+ *
+ *
+ */
 function useURLSort({ sortFields, defaultSortOption, onSort }: UseURLSortProps): UseURLSortResult {
     const [sortOption, setSortOption] = useURLParameter('sortOption', defaultSortOption);
 
     // get the parsed sort option values from the URL, if available
     // otherwise, use the default sort option values
-    const activeSortField =
-        isParsedQs(sortOption) && typeof sortOption?.field === 'string'
-            ? sortOption.field
-            : defaultSortOption.field;
-    const activeSortDirection =
-        isParsedQs(sortOption) && isDirection(sortOption?.direction)
-            ? sortOption.direction
-            : defaultSortOption.direction;
-    const activeAggregateBy =
-        isParsedQs(sortOption) && sortOption?.aggregateBy
-            ? (sortOption?.aggregateBy as SortAggregate)
-            : undefined;
-
-    const internalSortResultOption = useRef<ApiSortOption>(
-        tableSortOption(activeSortField, activeSortDirection, activeAggregateBy)
-    );
-
-    // we'll use this to map the sort fields to an index PatternFly can use internally
-    const [fieldToIndexMap, setFieldToIndexMap] = useState<Record<string, number>>({});
-
-    // we'll construct a map of sort fields to indices that will make it easier to work with
-    // PatternFly
-    useDeepCompareEffect(() => {
```

Comment:
Instead of relying on an effect, we'll just map the fields to the PatternFly index directly in `getSortParams()` which is more readable and very likely more performant. Instead of a direct indexing into a map, we compare each item provided in `sortFields` to each item passed to `getSortParams` to determine if there is overlap, and if so, use the matching `sortFields` position as the index.